File: Esej.cpp
#include <iostream>
#include "bits/stdc++.h"
using namespace std;

void createVocab(vector<string> &words, int depth, string &currentWord) {
    if(depth == 4) {
        words.push_back(currentWord);
        return;
    }
    for (int i = 97; i < 123; ++i) {
        currentWord[depth] = i;
        createVocab(words, depth+1, currentWord);
    }
}

int main() {
    int a, b;
    cin >> a >> b;
    vector<string> words;
    string currentWord = "    ";
    createVocab(words, 0, currentWord);
    for (int i = 0; i < b; ++i) {
        cout << words[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Esej.py
a, b = map(int, input().split())
v_size = b//2


def create_vocab(current_vocab=[], depth=0, current_word=['' for _ in range(4)]):
    if depth == 4:
        current_vocab.append(''.join(current_word))
        return
    for i in range(97, 123):
        current_word[depth] = chr(i)
        create_vocab(current_vocab, depth+1, current_word)
    return current_vocab


vocab = create_vocab()
print(' '.join(vocab[:b]))
